packages/beacon-node/src/chain/validation/attestation.ts

```diff
@@ -510,27 +509,26 @@ async function validateAttestationNoSignatureCheck(
   }
 
   // no signature check, leave that for step1
-  const indexedAttestationContent = {
+  const indexedAttestation: IndexedAttestation = {
     attestingIndices,
     data: attData,
     signature,
   };
-  const indexedAttestation =
-    ForkSeq[fork] >= ForkSeq.electra
-      ? (indexedAttestationContent as electra.IndexedAttestation)
-      : (indexedAttestationContent as phase0.IndexedAttestation);
 
-  const attestationContent = attestationOrCache.attestation ?? {
-    aggregationBits,
-    data: attData,
-    committeeIndex,
-    signature,
-  };
-
-  const attestation =
-    ForkSeq[fork] >= ForkSeq.electra
-      ? (attestationContent as SingleAttestation<ForkPostElectra>)
-      : (attestationContent as SingleAttestation<ForkPreElectra>);
+  const attestation: SingleAttestation = attestationOrCache.attestation
+    ? attestationOrCache.attestation
+    : !isForkPostElectra(fork)
+      ? {
+          aggregationBits: aggregationBits as BitArray,
```

Comment:
Need to type cast here as `aggregationBits` is `null` post-electra, I don't think it's required to add another assert here

Reply:
Probably need a short comment here to clarify

Reply:
added comment